extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/runtime/SessionOperations.java

```diff
@@ -74,7 +109,8 @@ static <T> Uni<T> withSessionOnDemand(Supplier<Uni<T>> work) {
             return work.get().eventually(() -> {
                 context.removeLocal(SESSION_ON_DEMAND_KEY);
                 context.removeLocal(SESSION_ON_DEMAND_OPENED_KEY);
-                return closeSession();
+                // TODO Luca this is just wrong
+                return closeSession(DEFAULT_PERSISTENCE_UNIT_NAME);
```

Comment:
I suppose this should close all currently opened sessions, no?

Reply:
It's debatable.

My opinion is that this says "open a session automatically when asked", and then later (this is lazy), someone opens a session, at which point we know which session and which PU. 

If we only allow a single session open at all times, then this should close that session that we opened, no matter its PU.

If we allow any number of sessions for different PUs open within a `@WithSessionOnDemand` then they should all be closed.

Reply:
I still think this should close the lazily opened session, not the default one.